nucleus/core/kernel/src/main/java/org/glassfish/runnablejar/commandline/Option.java

```diff
@@ -0,0 +1,264 @@
+            + " \"" + Arguments.DEPLOY_KEY_PREFIX + "app=myapp.war\"."
+            + " The property \"properties\" can also be defined in this file, pointing to another file."
+            + " In that case, properties will be loaded also from that file.") {
+
+        public void handle(String value, Arguments arguments) {
+            loadPropertiesFromFile(value, arguments);
+        }
+    },
+    HTTP_PORT("httpPort", Set.of("port", "p"), "-p=PORT, --httpPort=PORT, --port=PORT",
+            "Bind the HTTP listener to the specified port. If not set, the HTTP listener binds to port"
+            + " 8080 by default, unless it's disabled by the --noListener argument.") {
+        @Override
+        public void handle(String value, Arguments arguments) {
+            setPort(DEFAULT_HTTP_LISTENER, value, arguments);
+        }
+    },
+    HTTPS_PORT("httpsPort", "--httpsPort=PORT_NUMBER",
+            "Bind the HTTPS listener to the specified port. If not set, the HTTPS listener is disabled by default.") {
+        @Override
+        public void handle(String value, Arguments arguments) {
+            setPort(DEFAULT_HTTPS_LISTENER, value, arguments);
+        }
+    },
+    DOMAIN_CFG_FILE("domainConfigFile", "--domainConfigFile=FILE",
+            "Set the location of domain configuration file (i.e., domain.xml) using which GlassFish should run.") {
+        @Override
+        public void handle(String value, Arguments arguments) {
+            arguments.glassFishProperties.setConfigFileURI(value);
+        }
+    },
+    DOMAIN_DIR("domainDir", Set.of("instanceRoot"), "--domainDir=DIRECTORY, --instanceRoot=DIRECTORY",
+            "Set the instance root (a.k.a. domain directory) using which GlassFish should run.") {
+        @Override
+        public void handle(String value, Arguments arguments) {
+            arguments.glassFishProperties.setInstanceRoot(value);
+        }
+    },
+    NO_LISTENER("noListener", Set.of("noPort"), "--noListener, --noPort",
+            "Disable the HTTP listener, which is by default enabled and bound to port 8080.") {
+        @Override
+        public void handle(String value, Arguments arguments) {
+            setPort(DEFAULT_HTTP_LISTENER, 0, arguments);
+        }
+    },
+    AUTO_DEPLOY_DIR("autoDeployDir", "--autoDeployDir=DIRECTORY",
+            "Files and directories in this directory will be deployed as applications (in random order), as if they"
+            + " were specified on the command line.") {
+        @Override
+        public void handle(String value, Arguments arguments) {
+            loadApplicationsFromDirectory(value, arguments);
+        }
+    },
+    LOG_LEVEL("logLevel", "--logLevel=LEVEL",
+            "Set the log level of all loggers to LEVEL") {
+        @Override
+        public void handle(String logLevel, Arguments arguments) {
+            setLogLevel(logLevel, arguments);
+        }
+    },
+    LOG_PROPERTIES("logProperties", "--logProperties=FILE",
```

Comment:
This will cause issues, because JDK loads first and also starts using the logging configuration, the only reliable option is to use -Djava.util.logging.config.file on command line.